### ISSUE TITLE
Updating a subdocument with Model.updateOne complains with "missing following properties of LeanDocument: ownerDocument, parent "

### ISSUE THREAD

**Issue by vkarpov15** [Automattic/mongoose#10597]
### Discussed in https://github.com/Automattic/mongoose/discussions/10595

<div type='discussions-op-text'>

<sup>Originally posted by **Duchynko** August 24, 2021</sup>
Mongoose version: 5.13.7

After an update from 5.11.17 to 5.13.7, I'm seeing this error when trying to update a list of subdocuments.
```
Type 'IGameEvent[]' is not assignable to type 'LeanDocument<IGameEventDocument>[]'.
  Type 'IGameEvent' is missing the following properties from type 'LeanDocument<IGameEventDocument>': ownerDocument, parent
```

```ts
// gameRecord: IGameDocument
gameRecord.update({
    status: GameStatus[fixture.status.short],
    awayTeamScore: goals.away,
    homeTeamScore: goals.home,
    venue: fixture.venue.name,
    date: new Date(fixture.date),
    // this is complaining. mapFixtureEvents is returning IGameEvent[] and events is expecting LeanDocument<IGameEventDocument>
    events: FootballApiMappings.mapFixtureEvents(events),
});
```

How can I update a subdocument with a plain object? 

```
// TypeScript doesn't complain here
gameRecord.events.push(FootballApiMappings.mapFixtureEvents(events));
```

Thanks :)
</div>

**Comment by vkarpov15** [comment]
Try upgrading to v5.13.9, it looks like this may have been fixed with #10647 . I can confirm the below script fails to compile with v5.13.7 but compiles successfully with v5.13.9.

```typescript
import { Document, connect, model, Schema } from 'mongoose';
  
run().catch(err => console.log(err));

async function run() {
  await connect('mongodb://localhost:27017/test');
  
  interface IGameEventDocument extends Document {
    description: string;
  }
  interface IGameDocument extends Document {
    name: string;
    events: IGameEventDocument[]
  }
  const schema = new Schema<IGameDocument>({ name: String, events: [{ description: String }] });

  const GameModel = model<IGameDocument>('Game', schema);

  const doc = await GameModel.findOne().orFail();
  await doc.update({ events: [{ description: 'test' }] });
}
```

**Comment by Duchynko** [comment]
@vkarpov15 In my case, `IGameEventDocument` extends `Subdocument` not `Document`.

```typescript
export type IGameEventDocument = IGameEvent & Types.Subdocument;
```

`events` field in update methods now expects `IGameEventDocument` instead of `LeanDocument<IGameEventDocument>`.

```typescript
const gameRecord = await Game.findByIdAndUpdate(upcomingGame._id, {
    status: GameStatus[fixture.status.short],
    awayTeamScore: goals.away,
    homeTeamScore: goals.home,
    venue: fixture.venue.name,
    date: new Date(fixture.date),
    // Type 'IGameEvent[]' is not assignable to type 'IGameEventDocument[]'.
    //  Type 'IGameEvent' is not assignable to type 'IGameEventDocument'.
    //   Type 'IGameEvent' is missing the following properties from type 'Subdocument': $isSingleNested, ownerDocument, parent, $parent, and 51 more.
    events: FootballApiMappings.mapFixtureEvents(events), // <- Error here
});
```

**Comment by vkarpov15** [comment]
@Duchynko Fix will be in v6.0.8. Are you able to upgrade to v6 or is there something that's blocking you from upgrading?

**Comment by Duchynko** [comment]
Thanks a lot @vkarpov15. I'll update to 6.0.8 👍🏼
